# Enable shared library build patch
# Copyright (C) 2010-2015 Georgi D. Sotirov <gdsotirov@dir.bg>
#
# This program is free software; you can redistribute it and/or
# modify it under the terms of the GNU General Public License
# as published by the Free Software Foundation; either version 2
# of the License, or (at your option) any later version.
#
# This program is distributed in the hope that it will be useful,
# but WITHOUT ANY WARRANTY; without even the implied warranty of
# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
# GNU General Public License for more details.
#
# You should have received a copy of the GNU General Public License
# along with this program; if not, write to the Free Software
# Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
#
diff -urNad lua-5.2.4-orig/Makefile lua-5.2.4/Makefile
--- lua-5.2.4-orig/Makefile	2015-02-24 03:28:54.000000000 +0200
+++ lua-5.2.4/Makefile	2015-05-02 13:14:10.000000000 +0300
@@ -41,7 +41,7 @@
 # What to install.
 TO_BIN= lua luac
 TO_INC= lua.h luaconf.h lualib.h lauxlib.h lua.hpp
-TO_LIB= liblua.a
+TO_LIB= liblua.a liblua.so
 TO_MAN= lua.1 luac.1
 
 # Lua version and release.
diff -urNad lua-5.2.4-orig/src/Makefile lua-5.2.4/src/Makefile
--- lua-5.2.4-orig/src/Makefile	2013-11-11 13:45:49.000000000 +0200
+++ lua-5.2.4/src/Makefile	2015-05-02 13:14:10.000000000 +0300
@@ -29,21 +29,30 @@
 PLATS= aix ansi bsd freebsd generic linux macosx mingw posix solaris
 
 LUA_A=	liblua.a
+LUA_SO= liblua.so
 CORE_O=	lapi.o lcode.o lctype.o ldebug.o ldo.o ldump.o lfunc.o lgc.o llex.o \
 	lmem.o lobject.o lopcodes.o lparser.o lstate.o lstring.o ltable.o \
 	ltm.o lundump.o lvm.o lzio.o
+CORE_PO=lapi.po lcode.po lctype.po ldebug.po ldo.po ldump.po lfunc.po lgc.po llex.po \
+	lmem.po lobject.po lopcodes.po lparser.po lstate.po lstring.po ltable.po \
+	ltm.po lundump.po lvm.po lzio.po
 LIB_O=	lauxlib.o lbaselib.o lbitlib.o lcorolib.o ldblib.o liolib.o \
 	lmathlib.o loslib.o lstrlib.o ltablib.o loadlib.o linit.o
+LIB_PO=	lauxlib.po lbaselib.po lbitlib.po lcorolib.po ldblib.po liolib.po \
+	lmathlib.po loslib.po lstrlib.po ltablib.po loadlib.po linit.po
 BASE_O= $(CORE_O) $(LIB_O) $(MYOBJS)
+BASE_PO=$(CORE_PO) $(LIB_PO) $(MYOBJS)
 
 LUA_T=	lua
 LUA_O=	lua.o
+LUA_PO= lua.po
 
 LUAC_T=	luac
 LUAC_O=	luac.o
+LUAC_PO=luac.po
 
-ALL_O= $(BASE_O) $(LUA_O) $(LUAC_O)
-ALL_T= $(LUA_A) $(LUA_T) $(LUAC_T)
+ALL_O= $(BASE_O) $(BASE_PO) $(LUA_O) $(LUA_PO) $(LUAC_O) $(LUAC_PO)
+ALL_T= $(LUA_A) $(LUA_SO) $(LUA_T) $(LUAC_T)
 ALL_A= $(LUA_A)
 
 # Targets start here.
@@ -55,16 +64,25 @@
 
 a:	$(ALL_A)
 
+%.o : %.c
+	$(CC) $(CFLAGS) -o $@ -c $<
+
+%.po : %.c
+	$(CC) $(CFLAGS) -fPIC -o $@ -c $<
+
 $(LUA_A): $(BASE_O)
 	$(AR) $@ $(BASE_O)
 	$(RANLIB) $@
 
-$(LUA_T): $(LUA_O) $(LUA_A)
+$(LUA_T): $(LUA_O) $(LUA_SO)
 	$(CC) -o $@ $(LDFLAGS) $(LUA_O) $(LUA_A) $(LIBS)
 
-$(LUAC_T): $(LUAC_O) $(LUA_A)
+$(LUAC_T): $(LUAC_O) $(LUA_SO)
 	$(CC) -o $@ $(LDFLAGS) $(LUAC_O) $(LUA_A) $(LIBS)
 
+$(LUA_SO): $(CORE_PO) $(LIB_PO)
+	$(CC) -o $@ -shared -lm -ldl $?
+
 clean:
 	$(RM) $(ALL_T) $(ALL_O)
 
@@ -103,7 +121,7 @@
 generic: $(ALL)
 
 linux:
-	$(MAKE) $(ALL) SYSCFLAGS="-DLUA_USE_LINUX" SYSLIBS="-Wl,-E -ldl -lreadline"
+	$(MAKE) $(ALL) SYSCFLAGS="-DLUA_USE_LINUX" SYSLIBS="-Wl,-E -ldl -lreadline -lhistory -lncurses" MYCFLAGS="$(MYCFLAGS)"
 
 macosx:
 	$(MAKE) $(ALL) SYSCFLAGS="-DLUA_USE_MACOSX" SYSLIBS="-lreadline" CC=cc
